pontoon/base/views.py

```diff
@@ -988,50 +993,75 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def get_download_content(slug, code, relative_path):
+    """
+    Get content for the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str relative_path: Resource path.
+    """
+    # Clone or update project repository
+    project = get_object_or_404(Project, slug=slug)
+    project.repositories.first().pull()
+
+    # Get source file path
+    locale = get_object_or_404(Locale, code__iexact=code)
+    vcs_project = VCSProject(project, locales=[locale])
+
+    source_path = relative_path
+    if relative_path.endswith('.po'):
+        source_path = relative_path + 't'
+    full_path = os.path.join(vcs_project.source_directory_path(), source_path)
```

Comment:
This is very scary because you're taking a path submitted by the user and loading it up, opening us up to things like users submitting paths like `../../../../../../../../../../../../etc/passwd`. Granted, we're attempting to parse the given file as some sort've resource file, so that particular example would fail, but whenever taking user input as a path you always have to ensure that the path is within some "safe" directory that you're limiting them to. In this case, we want to limit them to the checkout of the project in question.